### PR TITLE
(IAC-139) Convert module to Litmus test framework

### PR DESCRIPTION
**Pull request by lionce**
converted module to use litmus
add github actions
actions will be triggered when new change will be added to release

### CODE REVIEW

**Comment by lionce** [comment]
github actions report [here](https://github.com/puppetlabs/puppetlabs-vsphere/runs/347706213)

File: README.md [hunk 1]
```diff
@@ -332,6 +332,8 @@ The vSphere module is only available for Puppet Enterprise 3.7 and later. This m
 
 ## Development
 
+We are experimenting with a new tool for running acceptance tests. It's name is [puppet_litmus](https://github.com/puppetlabs/puppet_litmus) this replaces beaker as the test runner. To run the acceptance tests follow the instructions [here](https://github.com/puppetlabs/puppet_litmus/wiki/Tutorial:-use-Litmus-to-execute-acceptance-tests-with-a-sample-module-(MoTD)#install-the-necessary-gems-for-the-module).
```
Comment (michaeltlombardi) [inline README.md]: Is this still an experiment?

File: distelli-manifest.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+team-modules/puppetlabs-vsphere:
```
Comment (michaeltlombardi) [inline distelli-manifest.yml]: We're doing both pipelines and GHA for the time being, I take it?

File: spec/spec_helper_acceptance.rb [hunk 3]
```diff
@@ -1,192 +1,82 @@
...
-    Open3.popen2e(cmd) do |_stdin, stdout_err, wait_thr|
-      while line = stdout_err.gets
-        result[:output].push(line)
-        puts line
-      end
-      result[:exit_status] = wait_thr.value
-    end
-
-    result
+# frozen_string_literal: true
```
Comment (michaeltlombardi) [inline spec/spec_helper_acceptance.rb]: I _think_ we should use the litmus `configure!` method for our spec_helper_acceptance files now:

```ruby
require 'puppet_litmus'
require 'spec_helper_acceptance_local' if File.file?(File.join(File.dirname(__FILE__), 'spec_helper_acceptance_local.rb'))

PuppetLitmus.configure!
```

File: spec/spec_helper_local.rb [hunk 4]
```diff
@@ -1,4 +1,8 @@
 require 'rbvmomi'
```
Comment (michaeltlombardi) [inline spec/spec_helper_local.rb]: This file is labeled `spec_helper_local` but I _think_ convention is that it should be `spec_helper_acceptance_local` unless these helpers are also used for unit tests?

File: distelli-manifest.yml (hunk 2 above)
Comment (lionce) [reply distelli-manifest.yml]: It should be removed.

File: spec/spec_helper_local.rb (hunk 4 above)
Comment (lionce) [reply spec/spec_helper_local.rb]: You're right, i'll rename it 👍

File: README.md (hunk 1 above)
Comment (lionce) [reply README.md]: I'll rephrase this description 
Cheers!

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 require 'puppetlabs_spec_helper/module_spec_helper'
 require 'rspec-puppet-facts'
 
-require 'spec_helper_local' if File.file?(File.join(File.dirname(__FILE__), 'spec_helper_local.rb'))
+require 'spec_helper_acceptance_local' if File.file?(File.join(File.dirname(__FILE__), 'spec_helper_acceptance_local.rb'))
```
Comment (DavidS) [inline spec/spec_helper.rb]: this is not needed here

File: spec/acceptance/machine_spec.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 require 'spec_helper_acceptance'
+require 'spec_helper_acceptance_local'
```
Comment (DavidS) [inline spec/acceptance/machine_spec.rb]: this should not be needed, as `spec_helper_acceptance` is already requiring this

File: spec/acceptance/machine_delete_spec.rb [hunk 7]
```diff
@@ -1,5 +1,6 @@
 require 'spec_helper_acceptance'
 require 'securerandom'
+require 'spec_helper_acceptance_local'
```
Comment (DavidS) [inline spec/acceptance/machine_delete_spec.rb]: this should not be needed, as `spec_helper_acceptance` is already requiring this

File: spec/acceptance/machine_customization_spec.rb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 require 'spec_helper_acceptance'
 require 'securerandom'
+require 'spec_helper_acceptance_local'
```
Comment (DavidS) [inline spec/acceptance/machine_customization_spec.rb]: this should not be needed, as `spec_helper_acceptance` is already requiring this

File: spec/spec_helper_local.rb (hunk 4 above)
Comment (DavidS) [reply spec/spec_helper_local.rb]: this is still containing a lot of duplicated code to spec_helper_acceptance_local, please sort this out.

File: spec/spec_helper.rb [hunk 9]
```diff
@@ -1,7 +1,6 @@
 require 'puppetlabs_spec_helper/module_spec_helper'
 require 'rspec-puppet-facts'
-
-require 'spec_helper_local' if File.file?(File.join(File.dirname(__FILE__), 'spec_helper_local.rb'))
+require 'spec_helper_acceptance_local'
```
Comment (DavidS) [inline spec/spec_helper.rb]: `pdk update` will revert this:
```
diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index 01f9db1..feb5720 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -1,6 +1,7 @@
 require 'puppetlabs_spec_helper/module_spec_helper'
 require 'rspec-puppet-facts'
-require 'spec_helper_acceptance_local'
+
+require 'spec_helper_local' if File.file?(File.join(File.dirname(__FILE__), 'spec_helper_local.rb'))
 
 include RspecPuppetFacts 
```

spec_helper and spec_helper_local are for unit test setup&helpers, spec_helper_acceptance and spec_helper_acceptance_local are for acceptance test setup & helpers. Please do keep them separated. You can go back to the original files on master to figure out which helpers belong into which file.

**Comment by lionce** [comment]
with pdk update -> rubocop failed!
33 files inspected, 426 offenses detected
Working to fix them

**Comment by lionce** [comment]
- [+] - ignored rubocop issues
- [-] - need to fix the issue with spec_helper files. To be checked the spec_helper files.

File: spec/spec_helper.rb (hunk 9 above)
Comment (cmccrisken-puppet) [reply spec/spec_helper.rb]: Think this should be resolved now @DavidS . @lionce had done most of this before signing off.

**Comment by cmccrisken-puppet** [comment]
Current GHActions test run: https://github.com/puppetlabs/puppetlabs-vsphere/runs/472978262

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -0,0 +1,52 @@
+name: "release"
```
Comment (carabasdaniel) [inline .github/workflows/release.yml]: Please add the Honeycomb env-vars here

File: metadata.json [hunk 11]
```diff
@@ -62,7 +64,7 @@
       "version_requirement": ">= 4.7.0 < 7.0.0"
     }
   ],
-  "pdk-version": "1.15.0",
-  "template-url": "https://github.com/puppetlabs/pdk-templates#1.15.0",
-  "template-ref": "tags/1.15.0-0-g0bc522e"
+  "pdk-version": "1.16.0",
```
Comment (carabasdaniel) [inline metadata.json]: This should be updated to pdk 1.17

**Comment by cmccrisken-puppet** [comment]
**Github Actions Test Run:** https://github.com/puppetlabs/puppetlabs-vsphere/runs/485461469

**Comment by cmccrisken-puppet** [comment]
**Debian based OSs Docker acceptance tests:**
```shell
bundle exec rake litmus:acceptance:parallel
┌ [✔] Running against 5 targets.
├── [✔] localhost:2222, litmusimage/debian:8
├── [✔] localhost:2223, litmusimage/debian:9
├── [✔] localhost:2224, litmusimage/debian:10
├── [✔] localhost:2225, litmusimage/ubuntu:16.04
└── [✔] localhost:2226, litmusimage/ubuntu:18.04
================
localhost:2225, litmusimage/ubuntu:16.04
....................................................................................................

Finished in 41 minutes 22 seconds (files took 2.55 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 14126 exit 0
================
localhost:2222, litmusimage/debian:8
....................................................................................................

Finished in 43 minutes 57 seconds (files took 2.57 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 14111 exit 0
================
localhost:2223, litmusimage/debian:9
....................................................................................................

Finished in 44 minutes 28 seconds (files took 2.61 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 14115 exit 0
================
localhost:2226, litmusimage/ubuntu:18.04
....................................................................................................

Finished in 44 minutes 52 seconds (files took 2.61 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 14135 exit 0
================
localhost:2224, litmusimage/debian:10
....................................................................................................

Finished in 45 minutes 6 seconds (files took 2.58 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 14131 exit 0
Successful on 5 nodes: ["localhost:2225, litmusimage/ubuntu:16.04", "localhost:2222, litmusimage/debian:8", "localhost:2223, litmusimage/debian:9", "localhost:2226, litmusimage/ubuntu:18.04", "localhost:2224, litmusimage/debian:10"]
```

**Comment by cmccrisken-puppet** [comment]
**RHEL based OSs Docker acceptance tests:**
```shell
bundle exec rake litmus:acceptance:parallel
┌ [✔] Running against 6 targets.
├── [✔] localhost:2222, litmusimage/centos:6
├── [✔] localhost:2223, litmusimage/oraclelinux:6
├── [✔] localhost:2224, litmusimage/scientificlinux:6
├── [✔] localhost:2225, litmusimage/centos:7
├── [✔] localhost:2226, litmusimage/oraclelinux:7
└── [✔] localhost:2227, litmusimage/scientificlinux:7
================
localhost:2227, litmusimage/scientificlinux:7
....................................................................................................

Finished in 38 minutes 17 seconds (files took 6.2 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 13285 exit 0
================
localhost:2223, litmusimage/oraclelinux:6
....................................................................................................

Finished in 39 minutes 42 seconds (files took 6.19 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 13273 exit 0
================
localhost:2222, litmusimage/centos:6
....................................................................................................

Finished in 43 minutes 18 seconds (files took 6.24 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 13275 exit 0
================
localhost:2225, litmusimage/centos:7
....................................................................................................

Finished in 47 minutes 2 seconds (files took 6.32 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 13254 exit 0
================
localhost:2226, litmusimage/oraclelinux:7
....................................................................................................

Finished in 47 minutes 6 seconds (files took 6.21 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 13256 exit 0
================
localhost:2224, litmusimage/scientificlinux:6
....................................................................................................

Finished in 47 minutes 14 seconds (files took 6.22 seconds to load)
100 examples, 0 failures

Libhoney::Client: no writekey configured, disabling sending events
pid 13267 exit 0
Successful on 6 nodes: ["localhost:2227, litmusimage/scientificlinux:7", "localhost:2223, litmusimage/oraclelinux:6", "localhost:2222, litmusimage/centos:6", "localhost:2225, litmusimage/centos:7", "localhost:2226, litmusimage/oraclelinux:7", "localhost:2224, litmusimage/scientificlinux:6"]
```

**Pull request merged**